### PR TITLE
Course selection test

### LINKED ISSUE #497: Automated UI test for course selection
Section 4 in TESTING.md

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Description of the PR

+ Closes #497 
+ Sometimes fails because of #244 
+ The RemoteRobot didn't like when the IDE was restarted so I skipped that part

# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt [hunk 1]
```diff
@@ -15,8 +19,45 @@ class PlaceholderTest {
...
+  }
+
+  @Test
+  fun turnProjectIntoAPlusProjectTest() = uiTest {
+    CommonSteps(this).openAPlusProjectWindow()
+    step("Cancel") {
+      with(dialog("Select Course")) {
+        button("Cancel").click()
+      }
+    }
```
Comment (OlliKiljunen) [inline src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt:31]: As far as I understand, this test requires that `simpleTest` was run (immediately?) before it? How we are going to maintain that order?

**Review by OlliKiljunen — APPROVED** [review]
Excellent!

What I'm only wondering now is how should we enforce the order of the tests. It seems to be [possible with JUnit](https://www.baeldung.com/junit-5-test-order) but it's different in versions 4 and 5. So, perhaps we should only implement that after upgrading to JUnit 5 in #553.

Until that, it's perhaps wisest that there's only one test procedure, or what do you think? So you can remove `simpleTest` and include line `CommonSteps(this).createProject()` in the beginning of your test.

File: src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt (hunk 1 above)
Comment (jaakkonarhi) [reply src/e2e/kotlin/fi/aalto/cs/apluscourses/e2e/PlaceholderTest.kt:31]: let's have just one test for now

**Comment by jaakkonarhi** [comment]
@superseacat I think a better option is to fix #244

**Comment by superseacat** [comment]
> @superseacat I think a better option is to fix #244

most likely it is, however, the single glitch in the process should not fail the whole process, so this idea (run `n` times) can be considered as a standalone as well 😉

**Comment by nikke234** [comment]
> most likely it is, however, the single glitch in the process should not fail the whole process, so this idea (run n times) can be considered as a standalone as well

This is very hacky even for my taste :smile:

**Review by Taikelenn — APPROVED** [review]

**Pull request merged**